[release] Documentation: release schedule (#58981)
woocommerce/woocommerce: docs/contribution/releases/schedule.md

---
post_title: WooCommerce Release Schedule
sidebar_label: Release Schedule
---

# WooCommerce Release Schedule

The schedule can be found [here](https://developer.woocommerce.com/release-calendar/): the page also explains the types 
of events in the calendar like `Releases`, `Release Candidates (RC)` and `Feature Freeze Dates`.

While the calendar reflects the events made public, there are specific steps in the release process that are internal, 
and this page aims to provide necessary context about those events for release leads and other involved parties.

## Detailed release schedule

This section will use the publicly available release schedule as anchors and clarify where the internal events fit in.

### Feature Freeze (start of the release cycle)

> This step is mostly automated, and in nut-shell, it creates a dedicated release branch where the future release undergoes testing and stabilization.

Reviewer: ```suggestion
This step is mostly automated, and in a nut-shell, it creates a dedicated release branch where the future release undergoes testing and stabilization.
```

Author: Updated in cee81db7ca87a22cbaa66e80b7a5526cf3461f40

At this point, the Developer Advocacy team publishes pre-release updates ([example](https://developer.woocommerce.com/2025/05/12/woocommerce-9-9-pre-release-updates/))

> ### RC1 (Feature Freeze + 1 week)

Reviewer: I really like these relative dates.  They're helpful in quickly understanding how the pieces of release fit together ❤️

This step is where various internal testing processes are happening: regression testing with canonical extensions,
regression testing in multiple environments, and exploration testing (incl. by the contributing teams).

### RC2 (RC1 + 2 weeks)

> TODO: not sure, thought is where we communicate RC availability for testing by community, but not sure.

Author: @jorgeatorres @prettyboymp : is my understanding here correct/up-to-date?

Reviewer: I think we describe this as the "anticipated final release candidate."  Meaning it _should_ be the last, but that is not a guarantee.

I think testing by the community could happen even earlier.  I'm not sure why we'd start testing at the code freeze build, but others would need to wait until RC2.  Plus as this point, it's getting pretty late to adopt any fixes into the release.

Reviewer 2: We communicate RC availability starting with RC-1. As @joshuatf said, there's no guarantee that RC-2 will be the last or that it will be exactly 1 week after RC-1, but normally that's the case. So it wouldn't be "RC1 + 2 weeks" but "Code Freeze + 2 weeks" under normal circumstances.

Author: Gotcha, updating.

Author: Here we go: d4658eaa71253334d1a660c9776a914a66de2752

> ### Release (aka Final release, RC2 + 1 week)

Reviewer: Could we just directly call this "Final Release?"  I think it removes ambiguity when referencing it.

```suggestion
### Final Release (RC2 + 1 week)
```

Author: Updated in ab2dc87fd16d4764d692ad023f5499046e8f5281

The final release process includes an additional staging step when it is not marked as stable yet and is deployed to our staging infrastructure.
If the staging step reveals critical issues, a dot-release will be created and follow the same staging procedure. Once staging is successful, the 
release is marked as stable and becomes available to everyone.

> At this point, the Developer Advocacy team publishes prepared in-advance release highlights ([example](https://developer.woocommerce.com/2025/06/09/woocommerce-9-9-its-fast-period/))

Reviewer: ```suggestion
At this point, the Developer Advocacy team publishes release highlights that are prepared in-advance ([example](https://developer.woocommerce.com/2025/06/09/woocommerce-9-9-its-fast-period/))
```

Noting that the highlights process may change, but obviously has not yet.

Author: Updated in ebf0fec4ae1baca669314385ccd58af291518000.

> Noting that the highlights process may change, but obviously has not yet.
:+1:, I think we'll need to review checklists and the docs one more time when new teams join the RL rotation.

## Delays

Due to business needs, the release dates may be subject to change. Here are some hints on approaching and effectively wrangling this situation.

> Once the need for changes in the release schedule is confirmed, create an internal post and communicate the necessary details.

Reviewer: >create an internal post and communicate the necessary details

Can we link to instructions on this?  I know some of this may be internal, but we'll need to figure out a way to link to internal resources somehow.

Author: Based on conversation here + p1750669722215689-slack-C07JJG089M0: updated in fa3353f653fe000d65f53dd64c6109eff75163d4

This post provides an opportunity for teams to share additional context, which may help invalidate and correct the schedule changes.

> Once the post feedback and release schedule changes cleared:

Reviewer: ```suggestion
Once the post feedback and release schedule changes have cleared:
```

Author: :+1:, updated in 244fcf3c65080972b4550d1b1ca8ebf473109502

> - in cooperation with the Developer Advocacy team, communicate the changes ([example](https://developer.woocommerce.com/2025/06/02/woocommerce-9-9-release-is-delayed/))

Reviewer: I think we should clarify who handles this.  Can't quite tell what the responsibilities are of the release lead vs dev ad.

Author: :+1:, updated the wording to clarify the responsibilities in 94fa6c005a3b0e8bccb333a794893569147440f2

> - actualize [the calendar](https://developer.woocommerce.com/release-calendar/)

Reviewer: ```suggestion
- Update [the calendar](https://developer.woocommerce.com/release-calendar/) with the new release dates
```

Author: Updated in 4cdf9d07938521fcebb03ea487326e78645cf37a

> Note: To minimize friction for teams, it's recommended not to change intervals between RCs and the final release but 

> > instead to shift the entire release cycle. The intervals consider multiple factors, including teams capacity.

Reviewer: ```suggestion
> instead to shift the entire release cycle. The intervals consider multiple factors, including the capacity across teams.
```

Author: Interesting construction `capacity across teams`, not in my active grammar constructs, but somehow makes sens to me.
Updated in c1affcdd38527c926b1ae53e54997ac79647f66b.

---
post_title: WooCommerce Release Schedule
sidebar_label: Release Schedule
---

# WooCommerce Release Schedule

The schedule can be found on [this page](https://developer.woocommerce.com/release-calendar/): the page also explains the types 
of events in the calendar like `Releases`, `Release Candidates (RC)` and `Feature Freeze Dates`.

> While the calendar reflects the events made public, there are specific steps in the release process that are internal, 

Reviewer 3: Nice

### Final Release (RC2 + 1 week)

This step is where the stable release version becomes available to everyone.

At this point, the Developer Advocacy team publishes release highlights that are prepared in-advance ([example](https://developer.woocommerce.com/2025/06/09/woocommerce-9-9-its-fast-period/)).

## Delays

Due to business needs, the release dates may be subject to change. Here are some hints on approaching and effectively wrangling this situation.

Once the need for changes in the release schedule is confirmed, create an internal Slack thread and communicate the necessary details.
This thread provides an opportunity for teams to share additional context, which may help invalidate and correct the schedule changes.

Once the thread feedback and release schedule changes have cleared:

- ask the Developer Advocacy team to communicate the changes publicly ([example](https://developer.woocommerce.com/2025/06/02/woocommerce-9-9-release-is-delayed/))
- update [the calendar](https://developer.woocommerce.com/release-calendar/) with the new release dates

> Note: To minimize friction for teams, it's recommended not to change intervals between RCs and the final release but 

> > instead to shift the entire release cycle. The intervals consider multiple factors, including the capacity across teams.

Reviewer 3: This feels only relevant internally.

---
post_title: WooCommerce Release Schedule
sidebar_label: Release Schedule
---

# WooCommerce Release Schedule

The schedule can be found on [this page](https://developer.woocommerce.com/release-calendar/): the page also explains the types 
of events in the calendar like `Releases`, `Release Candidates (RC)` and `Feature Freeze Dates`.

While the calendar reflects the events made public, there are specific steps in the release process that are internal, 
and this page aims to provide necessary context about those events for release leads and other involved parties.

## Detailed release schedule

This section will use the publicly available release schedule as anchors and clarify where the internal events fit in.

### Feature Freeze (start of the release cycle)

This step is mostly automated, and in a nut-shell, it creates a dedicated release branch where the future release undergoes testing and stabilization.
At this point, the Developer Advocacy team publishes pre-release updates ([example](https://developer.woocommerce.com/2025/05/12/woocommerce-9-9-pre-release-updates/))

### RC1 (Feature Freeze + 1 week)

This step is where various testing processes are happening:

> - internally: regression testing with canonical extensions, regression testing in multiple environments, and exploration testing (incl. by the contributing teams)

Reviewer 3: >(incl.

We should avoid abbreviations in docs

Author: :+1:, updated in 0d017503862a915ca17d584456f334fdd6a0bb89

> - community: the Developer Advocacy team updates the pre-release announcement so the community can give the RC a try

Reviewer 3: External users may not be familiar with the term "cannonincal extensions".

Author: Right. How does this look: 2fcb3088545a93e6e4d55134cd7a6a59b36341aa ?

Reviewer 3: Perfect, thank you!

> - community: the Developer Advocacy team updates the pre-release announcement so the community can give the RC a try

Reviewer 3: > the Developer Advocacy team updates the pre-release announcement so the community can give the RC a try

👇

> the Developer Advocacy team updates the pre-release announcement so the community can [begin testing](/docs/contribution/testing/beta-testing/) the release.

Author: Updated in 4f94460f3d9bba1c4177cb05a2c0129a65e2050c

### RC2 (RC1 + 1 week)

This step is where we release regressions fixes discovered in RC1, as part of the release stabilization.

At this point, the Developer Advocacy team continues to update the pre-release announcement.

### Final Release (RC2 + 1 week)

The final release process includes an additional staging step when it is not marked as stable yet and is deployed to our staging infrastructure.
If the staging step reveals critical issues, a dot-release will be created and follow the same staging procedure. Once staging is successful, the 
release is marked as stable and becomes available to everyone.

> At this point, the Developer Advocacy team publishes release highlights that are prepared in-advance ([example](https://developer.woocommerce.com/2025/06/09/woocommerce-9-9-its-fast-period/))

Reviewer 3: missing period

Author: Updated in ebc420b6532403c16804b32d5bb2e86266306069

## Delays

Due to business needs, the release dates may be subject to change. Here are some hints on approaching and effectively wrangling this situation.

Once the need for changes in the release schedule is confirmed, create an internal Slack thread and communicate the necessary details.
This thread provides an opportunity for teams to share additional context, which may help invalidate and correct the schedule changes.

> > Note: Please refer to Pf59Ax-10U-p2, before starting the thread.

Reviewer 3: This seems like maybe an internal note for engineering, but not something gives external folks any real value, since our slack channels and discussions are not public.  Also not sure what Pf59Ax-10U-p2 is meant to be

Author: > This seems like maybe an internal note for engineering

Correct. In this case, we link as it needs review; potentially, we can extract relevant bits directly into this documentation. I intended to minimize detours and keep things timeboxed, but I can attempt to extract relevant information into this document if it makes sense.

> Also not sure what Pf59Ax-10U-p2 is meant to be

It's an internal link. See `Public GitHub Repository Shorthand` in FG.

Author: Updated in 2f5e233d727326c080270297059025e8b5927b71

> > Note: Please refer to Pf59Ax-10U-p2 (shorthand, internal link), before starting the thread.

Reviewer 3: This should not be included here.

Author: Removed in 61b2381efc66bb422e6ef5755cae40d2a4bdd5da

> > The playbook is slightly dated, but it provides valuable insights for informed decision-making and effective outreach to key stakeholders.

Reviewer 3: Is the playbook made public anywhere?

Author: See https://github.com/woocommerce/woocommerce/pull/58981/files#r2161501521
